### PR TITLE
build: convert build-related scripts to POSIX shell (#172)

### PR DESCRIPTION
**Pull request by isaac-io**
Clean up scripts that are used throughout the build and use bash syntax in order to allow running them on platforms that don't have bash installed, since they don't really require any bash-specific features and can easily be converted to POSIX shell.

### CODE REVIEW

**Review by speedbmike — COMMENTED** [review]
I don't like the changes with <<EOF after "if", imho it's better to have them before if. 
Also de morgan instead of if ! makes it less readable. 
Other than that looks ok

File: tools/rocksdb_dump_test.sh [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# shellcheck disable=SC2148
+#!/bin/sh
```
Comment (mrambacher) [inline tools/rocksdb_dump_test.sh:1]: Why this change?

**Review by mrambacher — COMMENTED** [review]
I am not sure I understand the purpose of this change.  What platform does not have bash available for it that we want this change?  It also adds another set of changes that will have to be merged and maintained btw Speedb and RocksDB.

File: tools/rocksdb_dump_test.sh (hunk 1 above)
Comment (isaac-io) [reply tools/rocksdb_dump_test.sh:1]: So that it could be run directly without having to explicitly going through `sh` on the command line.

**Comment by isaac-io** [comment]
> I am not sure I understand the purpose of this change. What platform does not have bash available for it that we want this change? It also adds another set of changes that will have to be merged and maintained btw Speedb and RocksDB.

Alpine Linux, which is used in CI, uses BusyBox by default, not Bash. This is easily solvable there by simply installing Bash, but I wanted to support platforms where that might not be possible. Though I agree that it adds unnecessary maintenance burden, so as long as there's no concrete requirement we probably shouldn't do it. Closing this PR for now.
